test/Microsoft.Health.Dicom.Tests.Integration/Persistence/ExtendedQueryTagErrorStoreTests.cs

```diff
@@ -0,0 +1,156 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+using Microsoft.Health.Dicom.Core.Features.Store;
+using Microsoft.Health.Dicom.Tests.Common;
+using Microsoft.Health.Dicom.Tests.Common.Extensions;
+using Xunit;
+
+namespace Microsoft.Health.Dicom.Tests.Integration.Persistence
+{
+    public class ExtendedQueryTagErrorStoreTests : IClassFixture<SqlDataStoreTestsFixture>, IAsyncLifetime
+    {
+        private readonly IExtendedQueryTagErrorStore _extendedQueryTagErrorStore;
+        private readonly IExtendedQueryTagStore _extendedQueryTagStore;
+        private readonly IIndexDataStore _indexDataStore;
+        private readonly DateTime _definedNow;
+        private long _watermark;
```

Comment:
`_watermark` here works like a global variable, it couples methods together and make logic complicated. could we remove it?